yb-voyager/src/migassessment/sizing.go

```diff
@@ -115,416 +172,503 @@ func SizingAssessment() error {
 	// calculate time taken for sharded import
 	importTimeForShardedObjects, parallelVoyagerJobsSharded, err :=
 		calculateTimeTakenAndParallelJobsForImport(SHARDED_LOAD_TIME_TABLE, shardedObjects,
-			coresToUse.numCores.Float64, coresToUse.memPerCore.Float64)
+			finalSizingRecommendation.VCPUsPerInstance, finalSizingRecommendation.MemoryPerCore)
 	if err != nil {
 		SizingReport.FailureReasoning = fmt.Sprintf("calculate time taken for sharded data import: %v", err)
 		return fmt.Errorf("calculate time taken for sharded data import: %w", err)
 	}
+
 	sizingRecommendation := &SizingRecommendation{
 		ColocatedTables:                 fetchObjectNames(colocatedObjects),
 		ShardedTables:                   fetchObjectNames(shardedObjects),
-		VCPUsPerInstance:                coresToUse.numCores.Float64,
-		MemoryPerInstance:               coresToUse.numCores.Float64 * coresToUse.memPerCore.Float64,
-		NumNodes:                        numNodes,
-		OptimalSelectConnectionsPerNode: optimalSelectConnections,
-		OptimalInsertConnectionsPerNode: optimalInsertConnections,
+		VCPUsPerInstance:                finalSizingRecommendation.VCPUsPerInstance,
+		MemoryPerInstance:               finalSizingRecommendation.VCPUsPerInstance * finalSizingRecommendation.MemoryPerCore,
+		NumNodes:                        finalSizingRecommendation.NumNodes,
+		OptimalSelectConnectionsPerNode: finalSizingRecommendation.OptimalSelectConnectionsPerNode,
+		OptimalInsertConnectionsPerNode: finalSizingRecommendation.OptimalInsertConnectionsPerNode,
 		ParallelVoyagerJobs:             math.Min(float64(parallelVoyagerJobsColocated), float64(parallelVoyagerJobsSharded)),
-		ColocatedReasoning:              reasoning,
+		ColocatedReasoning:              getReasoning(finalSizingRecommendation, shardedObjects, colocatedObjects),
 		EstimatedTimeInMinForImport:     importTimeForColocatedObjects + importTimeForShardedObjects,
 	}
 	SizingReport.SizingRecommendation = *sizingRecommendation
 
 	return nil
 }
 
-func fetchObjectNames(dbObjects []SourceDBMetadata) []string {
-	var objectNames []string
-	for _, dbObject := range dbObjects {
-		objectNames = append(objectNames, dbObject.SchemaName+"."+dbObject.ObjectName)
+/*
+pickBestRecommendation selects the best recommendation from a map of recommendations based on certain criteria.
```

Comment:
Instead of saying "certain criteria" - Can we say that we are optimizing for the cores. And hence we chose the setup where the number of cores is less.

Reply:
done. Added the comment.